certbot/certbot/_internal/main.py

```diff
@@ -921,6 +922,62 @@ def update_account(config: configuration.NamespaceConfig,
     return None
 
 
+def fetch_account(config: configuration.NamespaceConfig,
+                   unused_plugins: plugins_disco.PluginsRegistry) -> Optional[str]:
+    """Fetch account info on the server.
+
+    :param config: Configuration object
+    :type config: configuration.NamespaceConfig
+
+    :param unused_plugins: List of plugins (deprecated)
+    :type unused_plugins: plugins_disco.PluginsRegistry
+
+    :returns: `None`
+    :rtype: None
+
+    """
+    # Portion of _determine_account logic to see whether accounts already
+    # exist or not.
+    account_storage = account.AccountFileStorage(config)
+    accounts = account_storage.find_all()
+
+    if not accounts:
+        return "Could not find an existing account to fetch."
+
+    acc, acme = _determine_account(config)
+    cb_client = client.Client(config, acc, None, None, acme=acme)
+
+    if not cb_client.acme:
+        raise errors.Error("ACME client is not set.")
+
+    regr = cb_client.acme.query_registration(acc.regr)
+
+    if config.verbose_count > 0:
+        display_util.notify(f"Account URI: {regr.uri}")
+
+        thumbprint = jose.b64encode(regr.body.key.thumbprint(
+            hash_function=SHA256)).decode()
+        display_util.notify(f"Account thumbprint: {thumbprint}")
+
+    phones = []
+    emails = []
+
+    for contact in regr.body.contact:
+        if contact.startswith('tel:'):
+            phones.append(contact[4:])
+        if contact.startswith('mailto:'):
+            emails.append(contact[7:])
+
+    display_util.notify("Phone number{} associated with account: {}".format(
+                            "s" if len(phones) > 1 else "",
+                            ", ".join(phones) if len(phones) > 0 else "none"))
+    display_util.notify("Email address{} associated with account: {}".format(
+                            "es" if len(emails) > 1 else "",
+                            ", ".join(emails) if len(emails) > 0 else "none"))
+
+    return None
```

Comment:
What do you think about including the current value of `--server` somewhere in the output?

For a server like Boulder it's fairly redundant, because the server can be inferred from the Account URI. However, I'm not sure that is always the case. IIRC some of Sectigo's endpoints might not be so obvious?

Something like this:

    $ certbot fetch_account --staging
    Saving debug log to /var/log/letsencrypt/letsencrypt.log
    Registered account for server https://acme-staging-v02.api.letsencrypt.org/directory:
      Account URI: https://acme-staging-v02.api.letsencrypt.org/acme/acct/37176138
      Email contacts: none

Reply:
Shouldn't be too hard :wink: